e2e/fuzz/replay/replay_message_reader.go

```diff
@@ -135,28 +160,59 @@ func (r *replayMessageReader) startChunkReading(messagesChannel chan []*ReplayMe
 	}()
 }
 
-// checks if given node is done with execution of messages
-func (r *replayMessageReader) checkIfDoneWithExecution(validatorId pbft.NodeID, msg *pbft.MessageReq) {
-	//when nodes reach sequence that is higher than the sequence in file,
-	//or if it is in constant round change we know we either reached the end of execution or nodes are stuck and there is a problem
-	if msg.View.Sequence > r.lastSequence ||
-		isSequenceInContinuousRoundChange(msg, validatorId) {
+// processMessage checks if message is a timeout, or if all messages have been read from the queue
+func (r *replayMessageReader) processMessage(validatorId pbft.NodeID, msg *pbft.MessageReq) bool {
+	if isTimeoutMessage(msg) {
+		return true
+	}
+
+	if msg.View.Sequence > r.lastSequenceMessages[validatorId].sequence ||
+		(msg.View.Sequence == r.lastSequenceMessages[validatorId].sequence && r.areMessagesFromLastSequenceProcessed(msg, validatorId)) {
 		r.lock.Lock()
-		if _, isDone := r.nodesDoneWithExecution[validatorId]; !isDone {
+		_, isDone := r.nodesDoneWithExecution[validatorId]
+		if !isDone {
```

Comment:
Why did you give up on this pattern? Code looks more compact that way (at least to me) :slightly_smiling_face: 
```suggestion
		if _, isDone := r.nodesDoneWithExecution[validatorId]; !isDone {
```